src/diffusers/models/unet_2d_blocks.py

```diff
@@ -587,7 +587,7 @@ def __init__(
         self.attentions = nn.ModuleList(attentions)
         self.resnets = nn.ModuleList(resnets)
 
-    def forward(self, hidden_states, temb=None):
+    def forward(self, hidden_states, temb=None, *args, **kwargs):
```

Comment:
```suggestion
    def forward(self, hidden_states, temb=None):
```
Is this really needed?

Reply:
Can we remove this, I don't see exactly how this is required here

Reply:
``` # 4. mid
        if self.mid_block is not None:
            sample = self.mid_block(
                sample,
                emb,
                encoder_hidden_states=encoder_hidden_states,
                attention_mask=attention_mask,
                cross_attention_kwargs=cross_attention_kwargs,
                encoder_attention_mask=encoder_attention_mask,
            )```
         
 These lines make the *args, **kwargs necessary. Unless I should add a ``` has_attention``` attribute and work off of that, like the Downsampling and upsampling blocks?

Reply:
```suggestion
            if self.mid_block is not None:
            if hasattr(self.mid_block, "has_cross_attention") and self.mid_block.has_cross_attention:
                sample = self.mid_block(
                    sample,
                    emb,
                    encoder_hidden_states=encoder_hidden_states,
                    attention_mask=attention_mask,
                    cross_attention_kwargs=cross_attention_kwargs,
                    encoder_attention_mask=encoder_attention_mask,
                )
            else:
                sample = self.mid_block(
                    sample,
                    emb,
                )
```
Would following your suggestion above and then doing this work, @patrickvonplaten?

Reply:
This sounds good to me!

Reply:
All right, I'll commit the changes. Just wanted to check once.